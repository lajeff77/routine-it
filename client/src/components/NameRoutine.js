import React from 'react';
import {Box, Card, TextField, Typography} from '@mui/material';

const NameRoutine = () => {
    //TODO: validate that the name isn't a duplicate
    return(
        <Box sx={{margin:"20px", display:"flex", flexDirection:"column", alignItems:"center"}}>
            <Typography variant="h1" sx={{textAlign: 'center', margin: '30px'}}>Let's Make a New Routine!</Typography>
            <Card variant="outlined" sx={{align:"center", width:"70%", display:"flex", flexDirection:"column", outline:"2px"}}>
                <Typography variant="body1" sx={{margin: '20px'}}>We'll start with a name for this routine:</Typography>
                <TextField required id="outlined-required" label="Enter a name" defaultValue="New Routine" variant="outlined" sx={{margin: '20px'}}/>
            </Card>
        </Box>
    )
}

export default NameRoutine;