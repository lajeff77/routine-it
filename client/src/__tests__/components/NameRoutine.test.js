import { render } from "@testing-library/react";
import {expect, test} from '@jest/globals';
import React from "react";
import NameRoutine from "../../components/NameRoutine";

const mockFormik = {
    values: {
        name: '',
        tasks: [],
        idealRoutine: [],
        averageRoutine: [],
        survivalRoutine: []
    },
    errors: {},
    touched: {},
    validateForm: jest.fn(),
    handleChange: jest.fn(),
    handleSubmit: jest.fn(),
};

const testProps = {
    formik: mockFormik,
    handleSubmit: jest.fn()
};

describe(NameRoutine.name, ()=>{
    test('NameRoutine matches snapshot', () => {
        const {container} = render(<NameRoutine props={testProps}/>);
        expect(container.firstChild).toMatchSnapshot();
    });
});