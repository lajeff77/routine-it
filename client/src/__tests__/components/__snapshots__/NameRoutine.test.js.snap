// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameRoutine NameRoutine matches snapshot 1`] = `
<div
  class="MuiBox-root css-2u9sdq"
>
  <h1
    class="MuiTypography-root MuiTypography-h1 css-3ns5jg-MuiTypography-root"
  >
    Let's Make a New Routine!
  </h1>
  <div
    class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1twd6hf-MuiPaper-root-MuiCard-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-altc5y-MuiTypography-root"
    >
      We'll start with a name for this routine:
    </p>
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="outlined-required"
        id="outlined-required-label"
      >
        Enter a name
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
          id="outlined-required"
          name="name"
          required=""
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-yjsfm1"
          >
            <span>
              Enter a name
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
